src/dashboard/src/components/api/views.py

```diff
@@ -819,16 +811,30 @@ def package(request):
 
 def _package_create(request):
     """Create a package."""
+    MESSAGE = "Package cannot be created"
     try:
         payload = json.loads(request.body)
         path = base64.b64decode(payload.get("path"))
     except (TypeError, ValueError):
         return helpers.json_response(
             {"error": True, "message": 'Parameter "path" cannot be decoded.'}, 400
         )
+    transfer_type = payload.get("type")
+    try:
+        if transfer_type not in amtypes.TRANSFER_TYPES and transfer_type is not None:
+            raise (
+                ValueError(
+                    "Unexpected type of package provided '{}'".format(transfer_type)
+                )
+            )
+    except ValueError as err:
+        LOGGER.error("{}: {}".format(MESSAGE, err))
+        return helpers.json_response(
+            {"error": True, "message": "{}: {}".format(MESSAGE, err)}, 500
+        )
```

Comment:
What would be the value of using try/catch when the error that you're catching is the one that you're raising? It sounds like it'd be much simpler to write something like `if condition: do X; do Y`.